Mathlib/Combinatorics/Matroid/Closure.lean

```diff
@@ -644,7 +644,8 @@ alias Indep.not_mem_closure_diff_of_mem := Indep.notMem_closure_diff_of_mem
 lemma Indep.closure_insert_diff_eq_of_mem_closure (hI : M.Indep I) (hf : f ∈ M.closure I)
     (he : e ∈ M.closure (insert f I \ {e})) : M.closure (insert f I \ {e}) = M.closure I := by
   apply subset_antisymm <;> apply closure_subset_closure_of_subset_closure
-  · rintro a (rfl | haI)
+  · simp only [subset_def, mem_diff, mem_insert_iff, mem_singleton_iff]
```

Comment:
Can you comment on this change? Does it fix hidden defeq abuse, is unintentional, something else?